pkg/server/plugin/datastore/sql/migration.go

```diff
@@ -35,25 +65,81 @@ func migrateDB(db *gorm.DB, dbType string, log hclog.Logger) (err error) {
 	if err := db.Assign(Migration{}).FirstOrCreate(migration).Error; err != nil {
 		return sqlError.Wrap(err)
 	}
-	version := migration.Version
 
-	if version > codeVersion {
-		err = sqlError.New("backwards migration not supported! (current=%d, code=%d)", version, codeVersion)
-		log.Error(err.Error())
-		return err
+	schemaVersion := migration.Version
+
+	var dbCodeVersion semver.Version
+	// Auto-migrate can only be safely disabled if 0.9.0 migrations have already been run
+	// We know if 0.9.0 migrations have been run because dbCodeVersion will be populated if so
+	// TODO: related epic https://github.com/spiffe/spire/issues/1083
+	// This check can be removed in 0.10.0
+	if migration.CodeVersion == "" {
+		if disableMigration {
+			err := sqlError.New("must upgrade from version %s or higher to disable auto-migration", minimumCodeVersionToDisableMigrate)
```

Comment:
`minimumCodeVersionToDisableMigrate` will not change between release versions (except when eventually removed), so not using logging fields